fix: prevent IPs from being used as long URL

seaerchin opened the pull request:
## Problem
closes https://linear.app/ogp/issue/ISOM-2009/vapt-low-internal-and-private-ip-addresses-as-long-urls

## Solution
- validate that the domain section of the url is not an IP address
- this validator is used across the dashboard, api and bulk upload 
- also validated that this prevents users from changing existing links to have an IP address as the long url 
    - i'm ok w/ the case where users **cannot** change existing long urls that are IP addresses but i haven't validated teh scale of the problem

seaerchin committed:
fix: prevent IPs from being used as long URL

dcshzj reviewed (approved):
Approving first, but I think Copilot got valid points regarding the test coverage, would be good to add.

seaerchin committed:
chore: add test for ip addresses

seaerchin committed:
chore: add try/catch loop

we do this so that if there's a drift in validation we can still return
a result

seaerchin committed:
chore: npm i
